lesson36/src/js/article.js

```diff
@@ -0,0 +1,114 @@
+const articleList = document.getElementById("js-article-List");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+};
+
+const createErrorMessage = (text) => {
+  const errorMessage = createElementWithClassName("p", "article-error-message");
+  errorMessage.textContent = text;
+  return errorMessage;
+};
+
+const renderLoading = () => {
+  const loadingWrapper = createElementWithClassName("div", "article-loading-wrapper");
+  const loading = document.createElement("img");
+  loadingWrapper.id = "js-loading";
+  loading.src = "./img/article-loading.gif";
+  articleList.appendChild(loadingWrapper).appendChild(loading);
+};
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (endpoint) => {
+  const response = await fetch(endpoint);
+  if (!response.ok) {
+    const responseMessage = `${response.status}:${response.statusText}`;
+    articleList.append(createErrorMessage(responseMessage));
+    console.error(responseMessage);
+    return;
+  }
+  return await response.json();
+}
+
+const getArticleData = async (api) => {
+  try {
+    return await fetchData(api);
+  } catch (error) {
+    console.error(error);
+    articleList.before(createErrorMessage(error));
+  }
+}
+
+const renderArticleItems = (articleData) => {
+  articleData.forEach((data) => {
+    const item = createElementWithClassName("li", "article__item");
+    const content = createElementWithClassName("p", "article__item-content");
+    const author = document.createElement("p");
+    author.textContent = data.author;
+    content.textContent = data.quote;
+    articleList.appendChild(item).appendChild(author).after(content);
+  });
+}
+
+let totalPosts;
```

Comment:
IMO
I think It is not recommended to define a let outside of a function.  So how about implementing it this way?
[setter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set)
```js
const totalPosts = {
  set count(posts) {
    this.countedPosts.push(posts);
  },
  countedPosts: []
};
```
If we did it this way, it would not be possible to substitute from the outside.
```js
const init = async () => {
  renderLoading();
  const articleData = await getArticleData(endpointConfig.setEndpoint());
  removeLoading();
  if (articleData.data.length === 0) {
    articleList.append(createErrorMessage("表示する記事がありません"));
    return;
  }
//代入するのではなく totalPosts.countに対して値をセットしました。
  totalPosts.count = articleData.total;
  renderArticleItems(articleData.data);
  observeConfig.startObserve(articleList.lastElementChild, intersectHandler);
};

const intersectHandler = ([entry]) => {
  if (!entry.isIntersecting) return;
  observeConfig.stopObserve(entry.target);
  const articleItems = document.querySelectorAll(".article__item");
  if (articleItems.length < totalPosts.countedPosts) {
    renderLoading();
    setTimeout(getArticleDataAndUpdate, 500);
  }
};
```
[codesandbox](https://codesandbox.io/s/lesson36-forked-trji5i?file=/js/article.js)

Reply:
Thanks for the review!
I was able to learn about `getter / setter.` Thank you for the wonderful opportunity.
In this case, we decided that a simple object rather than a setter would be better since we are only assigning a value and returning it.Please check.
[d9f9b51](https://github.com/sae-github/JS_lesson/pull/47/commits/d9f9b51b7baeab88defc16714e1a10083b0fe7af)